File: backend/src/main/java/todoktodok/backend/reply/application/dto/response/ReplyResponse.java
package todoktodok.backend.reply.application.dto.response;

import todoktodok.backend.member.application.dto.response.MemberResponse;
import todoktodok.backend.reply.domain.Reply;

import java.time.LocalDateTime;

public record ReplyResponse(
        Long replyId,
        MemberResponse member,
        LocalDateTime createdAt,
        String content,
        int likeCount
) {

    public ReplyResponse(
            final Reply reply,
            final int likeCount
    ) {
        this(
                reply.getId(),
                new MemberResponse(reply.getMember()),
                reply.getCreatedAt(),
                reply.getContent(),
                likeCount
        );
    }
}

File: android/TodokTodok/data/src/main/java/com/team/data/network/response/comment/ReplyResponse.kt
package com.team.data.network.response.comment

import com.team.data.core.ext.toLocalDateTime
import com.team.data.network.response.discussion.MemberResponse
import com.team.data.network.response.discussion.toDomain
import com.team.domain.model.Reply
import kotlinx.serialization.Serializable

@Serializable
data class ReplyResponse(
    val content: String,
    val createdAt: String,
    val isLikedByMe: Boolean,
    val likeCount: Int,
    val member: MemberResponse,
    val replyId: Long,
)

fun ReplyResponse.toDomain() =
    Reply(
        content,
        createdAt.toLocalDateTime(),
        likeCount,
        member.toDomain(),
        replyId,
        isLikedByMe,
    )
